lib/iden3comm/domain/use_cases/get_auth_inputs_use_case.dart

```diff
@@ -54,34 +48,38 @@ class GetAuthInputsUseCase
 
   @override
   Future<Uint8List> execute({required GetAuthInputsParam param}) async {
-    IdentityEntity identity = await _getIdentityUseCase.execute(
-        param: GetIdentityParam(
-            genesisDid: param.genesisDid, privateKey: param.privateKey));
+    List<String> identityPublicKey = (await _getIdentityUseCase.execute(
+            param: GetIdentityParam(
+                genesisDid: param.genesisDid, privateKey: param.privateKey)))
+        .publicKey;
 
     List<String> authClaim =
-        await _getAuthClaimUseCase.execute(param: identity.publicKey);
-    NodeEntity authClaimNode =
-        await _identityRepository.getAuthClaimNode(children: authClaim);
+        await _getAuthClaimUseCase.execute(param: identityPublicKey);
+    String authClaimNodeHash =
+        (await _identityRepository.getAuthClaimNode(children: authClaim)).hash;
 
-    ProofEntity incProof = await _smtRepository.generateProof(
-        key: authClaimNode.hash,
-        type: TreeType.claims,
-        did: param.genesisDid,
-        privateKey: param.privateKey);
+    Map<String, dynamic> incProof = (await _smtRepository.generateProof(
+            key: authClaimNodeHash,
+            type: TreeType.claims,
+            did: param.genesisDid,
+            privateKey: param.privateKey))
+        .toJson();
```

Comment:
The UC is not the one changing the format of the data.
Either you pass the entity and the mapping is done in the Data layer, or you only pass the targeted data from the entity.